conduit/plugins/processors/filterprocessor/expression/expression.go

```diff
@@ -53,97 +47,74 @@ type regexExpression struct {
-	return e.Regex.MatchString(input.(string)), nil
+func (e *regexExpression) Search(input interface{}) (bool, error) {
+	switch v := input.(type) {
+	case string:
+		return e.Regex.MatchString(v), nil
+	default:
+		return false, fmt.Errorf("unexpected regex search input type (%T)", v)
+	}
 }
 
-// MakeExpression creates an expression based on an expression type
-func MakeExpression(expressionType FilterType, expressionSearchStr string, targetKind reflect.Kind) (*Expression, error) {
-	switch expressionType {
-	case ExactFilter:
-		{
-			r, err := regexp.Compile("^" + expressionSearchStr + "$")
-			if err != nil {
-				return nil, err
-			}
-
-			var exp Expression = regexExpression{Regex: r}
-			return &exp, nil
-		}
-	case RegexFilter:
-		{
-			r, err := regexp.Compile(expressionSearchStr)
-			if err != nil {
-				return nil, err
-			}
-
-			var exp Expression = regexExpression{Regex: r}
-			return &exp, nil
-		}
+func makeRegexExpression(searchStr string, expressionType FilterType) (Expression, error) {
+	if expressionType != EqualToFilter && expressionType != RegexFilter {
+		return nil, fmt.Errorf("target type (string) does not support %s filters", expressionType)
+	}
+	r, err := regexp.Compile(searchStr)
+	if err != nil {
+		return nil, err
+	}
+
+	return &regexExpression{Regex: r}, nil
+}
 
-	case LessThanFilter:
-		fallthrough
-	case LessThanEqualFilter:
-		fallthrough
-	case GreaterThanFilter:
-		fallthrough
-	case GreaterThanEqualFilter:
-		fallthrough
-	case EqualToFilter:
-		fallthrough
-	case NotEqualToFilter:
-		{
-			var exp Expression
-
-			switch targetKind {
-			case reflect.TypeOf(basics.MicroAlgos{}).Kind():
-				{
-					v, err := strconv.ParseUint(expressionSearchStr, 10, 64)
-					if err != nil {
-						return nil, err
-					}
-
-					exp = microAlgoExpression{
-						FilterValue: basics.MicroAlgos{Raw: v},
-						Op:          expressionType,
-					}
-				}
-
-			case reflect.Int64:
-				{
-					v, err := strconv.ParseInt(expressionSearchStr, 10, 64)
-					if err != nil {
-						return nil, err
-					}
-
-					exp = int64NumericalExpression{
-						FilterValue: v,
-						Op:          expressionType,
-					}
-				}
-
-			case reflect.Uint64:
-				{
-					v, err := strconv.ParseUint(expressionSearchStr, 10, 64)
-					if err != nil {
-						return nil, err
-					}
-
-					exp = uint64NumericalExpression{
-						FilterValue: v,
-						Op:          expressionType,
-					}
-				}
-
-			default:
-				return nil, fmt.Errorf("unknown target kind (%s) for filter type: %s", targetKind.String(), expressionType)
-			}
-
-			return &exp, nil
+func makeSignedExpression(searchStr string, expressionType FilterType) (Expression, error) {
+	if expressionType == RegexFilter {
+		return nil, fmt.Errorf("target type (numeric) does not support %s filters", expressionType)
+	}
+
+	v, err := strconv.ParseInt(searchStr, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("search string \"%s\" is not numeric: %w", searchStr, err)
```

Comment:
nit: maybe the error message can be something like "not a valid int64", because the err could be an int overflow error?

Reply:
Thanks. Updated, and added tests for those two cases.